### PR TITLE
Backbone implementation

### PR DESCRIPTION
**Pull request by darosior**
This implements the configuration parsing, bitcoind interface and database management as a foundation to then start implementing the actual watchtower logic.

### CODE REVIEW

File: src/bitcoind/interface.rs [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+                thread::sleep(Duration::from_secs(1));
+            }
+            jsonrpc::Error::Json(ref err) => {
+                log::error!(
+                    "JSON serialization error when talking to bitcoind: '{}'",
+                    err
+                );
+                // Weird. A JSON serialization error? Just try again but
+                // fail fast anyways as it should not happen.
+                log::error!(
```
Comment (JSwambo) [inline src/bitcoind/interface.rs:169]: nit: no need to log twice?

File: src/bitcoind/mod.rs [hunk 2]
```diff
@@ -0,0 +1,423 @@
...
+
+            complete_sanity_check(&server);
+            complete_network_check(&server);
+
+            // Read the second 'getblockchaininfo' req
+            let (mut stream, _) = server.accept().unwrap();
+            // We can hang for a bit..
+            thread::sleep(time::Duration::from_secs(2));
+            read_til_json_end(&mut stream);
+            // .. And we them an error, they will retry
```
Comment (JSwambo) [inline src/bitcoind/mod.rs:358]: nit: give them an error

File: src/config.rs [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+pub struct ScriptsConfig {
+    #[serde(deserialize_with = "deserialize_fromstr")]
+    pub deposit_descriptor: DepositDescriptor,
+    #[serde(deserialize_with = "deserialize_fromstr")]
+    pub unvault_descriptor: UnvaultDescriptor,
+    #[serde(deserialize_with = "deserialize_fromstr")]
+    pub cpfp_descriptor: CpfpDescriptor,
+}
+
+/// Static informations we require to operate
```
Comment (JSwambo) [inline src/config.rs:87]: nit: information

File: src/database/mod.rs [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+    db_path: &path::Path,
+    deposit_descriptor: &DepositDescriptor,
+    unvault_descriptor: &UnvaultDescriptor,
+    cpfp_descriptor: &CpfpDescriptor,
+    network: Network,
+) -> Result<(), DatabaseError> {
+    let timestamp = time::SystemTime::now()
+        .duration_since(time::UNIX_EPOCH)
+        .map(|dur| timestamp_to_u32(dur.as_secs()))
+        .expect("Computing time since expoch");
```
Comment (JSwambo) [inline src/database/mod.rs:141]: nit: epoch

File: src/main.rs [hunk 5]
```diff
@@ -1 +1,214 @@
...
+        "Reading or generating Noise key at '{:?}'",
+        noise_secret_path
+    );
+    let noise_secret = read_or_create_noise_key(&noise_secret_path).unwrap_or_else(|e| {
+        log::error!("Error reading or generating Noise key: '{}'", e);
+        process::exit(1);
+    });
+    log::info!(
+        "Using Noise key '{}'.",
+        NoisePubKey(curve25519::scalarmult_base(&curve25519::Scalar(noise_secret.0)).0)
```
Comment (JSwambo) [inline src/main.rs:187]: nit: why not return `Ok(NoisePubKey, NoisePrivKey)` with `read_or_create_noise_key` and avoid using scalarmult_base?

**Review by JSwambo — APPROVED** [review]
tAck  97f4db11754155918271ba5855568f6f40e13261

File: src/bitcoind/interface.rs (hunk 1 above)
Comment (darosior) [reply src/bitcoind/interface.rs:169]: Added a commit to #3 fixing it

File: src/main.rs (hunk 5 above)
Comment (darosior) [reply src/main.rs:187]: We'd need `scalarmult_base` anyways but what we can do is simply use the `public_key()` helper of `SecretKey`, added a commit doing that to #3.

**Pull request merged**